src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -626,6 +627,9 @@ InvokeHostFunctionOpFrame::doCheckValid(SorobanNetworkConfig const& config,
     if (hostFn.type() == HOST_FUNCTION_TYPE_UPLOAD_CONTRACT_WASM &&
         hostFn.wasm().size() > config.maxContractSizeBytes())
     {
+        mParentTx.pushSimpleDiagnosticError(
+            SCE_BUDGET, SCEC_EXCEEDED_LIMIT,
```

Comment:
We should also add errors for the failure cases in `TransactionFrame::validateSorobanResources` (that's where Fons ran into his issues).